samtranslator/model/apigateway.py

```diff
@@ -90,9 +92,10 @@ def make_auto_deployable(self, stage, openapi_version=None, swagger=None):
         # to prevent redeployment when API has not changed
 
         # NOTE: `str(swagger)` is for backwards compatibility. Changing it to a JSON or something will break compat
+        # NOTE: If extending the hash data, make sure to add delimiter between the previous data and the new data.
         hash_input = str(swagger)
         if openapi_version:
-            hash_input = hash_input + str(openapi_version)
+            hash_input = hash_input + self._X_HASH_DELIMITER + str(openapi_version)
```

Comment:
Put the hash inputs into an array (only adding openapi_version in the if), then use [join()](https://www.programiz.com/python-programming/methods/string/join).

Reply:
done